clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -2755,7 +2759,12 @@ raw2trace_t::append_delayed_branch(raw2trace_thread_data_t *tdata, app_pc next_p
                                                 tdata->delayed_branch.end());
                     tdata->delayed_branch_decode_pcs.pop_back();
                     tdata->delayed_branch_target_pcs.pop_back();
-                    break;
+                    // Start the iteration over.
+                    --instr_count;
+                    instr_index = instr_count - 1;
+                    next_instr_pc = branch_addr;
+                    i = static_cast<int>(tdata->delayed_branch.size());
```

Comment:
Why do we set `i` to delayed_branch.size() still? Since, in a way, there's one fewer delayed branch to process, shouldn't this be something smaller?

Reply:
Removed.  (The reason was b/c the vector was shrunk so i was in fact smaller.)

Reply:
Still removed.